morecantile/models.py

```diff
@@ -849,17 +915,44 @@ def _ul(self, *tile: Tile) -> Coords:
 
         matrix = self.matrix(t.z)
         res = self._resolution(matrix)
+        origin_x, origin_y = self._matrix_origin(matrix)
 
-        origin_x = (
-            matrix.pointOfOrigin[1] if self._invert_axis else matrix.pointOfOrigin[0]
-        )
-        origin_y = (
-            matrix.pointOfOrigin[0] if self._invert_axis else matrix.pointOfOrigin[1]
+        cf = 1
+        if matrix.variableMatrixWidths is not None:
+            cf = matrix.get_coalesce_factor(t.y)
+
+        return Coords(
+            origin_x + math.floor(t.x / cf) * res * matrix.tileWidth,
```

Comment:
This is wrong! I think the resolution should be changed

Reply:
Yep! 
<img width="1083" alt="Screenshot 2023-07-21 at 1 25 36 AM" src="https://github.com/developmentseed/morecantile/assets/10407788/6e059bb6-832e-47a5-98e6-e3a724165ff8">